packages/sqlite/src/main/java/org/sqlite/SQLiteJDBCLoader.java

```diff
@@ -199,9 +199,12 @@ private static boolean extractAndLoadLibraryFile(
 
             // Set executable (x) flag to enable Java to load the native library
             var file = extractedLibFile.toFile();
-            file.setReadable(true);
-            file.setWritable(true, true);
-            file.setExecutable(true);
+            var didSetReadable = file.setReadable(true);
+            assert didSetReadable;
+            var didSetWritable = file.setWritable(true, true);
+            assert didSetWritable;
+            var didSetExecutable = file.setExecutable(true);
+            assert didSetExecutable;
```

Comment:
some light assertions for lib unpacking, which won't make it into prod (when assertions are turned off) anyway